examples/datasets/plot_limo_data.py

```diff
@@ -33,146 +35,282 @@
-# response, as participants should be better at identifying these faces.
+# As expected, the difference between Face A and Face B are very small.
+#
 #
-# Create design matrix for linear regression. We'll use the information
-# contained in the ``limo_epochs.metadata``.
-design = limo_epochs.metadata.copy()
-design = design.assign(intercept=1)  # add intercept
+# Visualize effect of stimulus phase-coherence
+# --------------------------------------------
+#
+# Since phase-coherence
+# determined whether a face stimulus could be easily identified,
+# one could expect that faces with high phase-coherence should evoke stronger
+# activation patterns along occipital electrodes.
+
+phase_coh = limo_epochs.metadata['phase-coherence']
+# get levels of phase coherence
+levels = sorted(phase_coh.unique())
+# create labels for levels of phase coherence (i.e., 0 - 85%)
+labels = ["{0:.2f}".format(i) for i in np.arange(0., .90, .05)]
+
+# create dict of evokeds for each level of phase-coherence
+evokeds = {label: limo_epochs[phase_coh == level].average()
+           for level, label in zip(levels, labels)}
+
+# pick channel to plot
+electrodes = ['C22', 'B11']
+# create figures
+for electrode in electrodes:
+    fig, ax = plt.subplots(figsize=(8, 4))
+    plot_compare_evokeds(evokeds,
+                         axes=ax,
+                         ylim=dict(eeg=(-15, 15)),
+                         picks=electrode,
+                         cmap=("Phase coherence", "magma"))
+
+###############################################################################
+# As shown above, there are some considerable differences between the
+# activation patterns evoked by stimuli with low vs. high phase-coherence at
+# the chosen electrodes.
+#
+#
+# Prepare data for linear regression analysis
+# --------------------------------------------
+#
+# Before we test the significance of these differences using linear
+# regression, we'll interpolate missing channels that were
+# dropped during preprocessing of the data.
+# Furthermore, we'll drop the EOG channels (marked by the "EXG" prefix)
+# present in the data:
+
+limo_epochs.interpolate_bads(reset_bads=True)
+
+limo_epochs.drop_channels(['EXG1', 'EXG2', 'EXG3', 'EXG4'])
+
+###############################################################################
+# Define predictor variables and design matrix
+# --------------------------------------------
+#
+# To run the regression analysis,
+# we need to create a design matrix containing information about the
+# variables (i.e., predictors) we want to use for prediction of brain
+# activity patterns. For this purpose, we'll use the information we have in
+# ``limo_epochs.metadata``: phase-coherence and Face A vs. Face B.
+
+# name of predictors + intercept
+predictor_vars = ['face a - face b', 'phase-coherence', 'intercept']
+
+# create design matrix
+design = limo_epochs.metadata[['phase-coherence', 'face']].copy()
 design['face a - face b'] = np.where(design['face'] == 'A', 1, -1)
```

Comment:
FYI, coding the Face variable as -1, 1 (often called "sum coding") will yield a beta coefficient that is actually half of what most people think it represents (see the table in section "what are the key coding schemes?" in [this blog post](http://talklab.psy.gla.ac.uk/tvw/catpred/)). Coding as -0.5, 0.5 instead yields the result that most people want / think they're getting with sum coding.  This is a subtlety of interpretation that is probably not that important in the context of this example, since we don't actually go so far as to interpret the magnitude of the betas, but if you ever decide to revise/expand this example to do that, bear this point in mind.

Reply:
true, you're right, thanks for the link.